src/nupic/engine/RegionImpl.hpp

```diff
@@ -106,7 +106,7 @@ namespace nupic
      * expects to find this method. Caller gets ownership. 
      */
 
-    // static Spec* createSpec();
+    static Spec* createSpec();
```

Comment:
We shouldn't implement a dummy version of this here since users should never call `RegionImpl::createSpec`. You can leave the declaration without an implementation to force the subclasses to implement it though. But I'm not sure if that works with static functions.